packages/server/lib/socket-base.ts

```diff
@@ -546,6 +556,9 @@ export class SocketBase {
             // surfacing the error to the app in open mode
             cb({ error })
           })
+          .then(() => {
```

Comment:
is [`finally`](http://bluebirdjs.com/docs/api/finally.html) an option here?
```suggestion
          .finally(() => {
```

Reply:
I think finally could work here... the thing with finally you have to watch out for is that for bluebird, it breaks promise chains... `finally(() => {'result'}).then((arg)=>{ arg != 'result')`

Reply:
done